atlasdb-dbkvs/src/test/java/com/palantir/atlasdb/keyvalue/dbkvs/impl/LazyClosableIteratorTest.java

```diff
@@ -41,20 +41,19 @@
 
 public class LazyClosableIteratorTest {
 
-    @Test
+    @Test // QA-89231
     public void testInterruptsDontCauseLeaksMultiple() throws Exception {
         for (int i = 1; i <= 100; i++) {
             try {
                 // Because race conditions are hard...
-                testInterruptsDontCauseLeaks();
+                assertInterruptsDontCauseLeaks();
             } catch (Throwable e) {
                 throw new RuntimeException("Failure during run " + i, e);
             }
         }
     }
 
-    @Test // QA-89231
-    public void testInterruptsDontCauseLeaks() throws Exception {
+    private void assertInterruptsDontCauseLeaks() throws Exception {
```

Comment:
Can we use barrier here? I guess that would be better and we can do away with less that 1-- tasks.